README.md

```diff
@@ -1,9 +1,310 @@
+  REPORT_DIR="[path to report directory]" \
+  ROSA_ENV=stage \
+  ./osde2e test \
+  --configs rosa,stage,addon-suite \
+  --must-gather false \
+  --destroy-cluster false \
+  --skip-health-check true
+  ```
+ 
+  **Arguments:** 
+  - The `--configs` arg here maps to `$CONFIGS` environment var in the prow config, see description in [parameters](#parameters) section. 
+  - `--destroy-cluster`, `--skip-health-check` and `--must-gather` help shorten the time consumed by the test to run locally. 
+
+  **Environment variables:**
+  - See [parameters](#parameters) section for description of environment variables used. 
+
+Once the execution is complete, you can view the report in the defined `REPORT_DIR` directory.
+
+After the Test Harness has been validated to work as intended locally, this flow can be performed in a CI pipeline to test agaisnt OSD releases as described below.
+
+## Locally Running Your Test Harness
+1. Create `ADDON_TEST_HARNESSES` image: 
+   Build and push latest docker image i.e.
+    ```bash
+    sudo docker build . -t quay.io/<-- your test harness image-->
+    sudo docker push quay.io/<--your test harness image-->
+    ``` 
+   Use this test image as the `ADDON_TEST_HARNESSES` in the next steps.
+2. Follow the steps in the [example above](#locally-running-this-example). Remember to change `ADDON_TEST_HARNESSES` as well as `ADDON_IDS` to your addon. 
+ 
+## Configuring OSDe2e
+
+1. Write addon test harness using this harness as an example.
+2. Build and push the latest docker image to a quay repo.
+3. Add a new prow job config in [osde2e periodic pipeline in release repo](https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/osde2e/openshift-osde2e-main-periodics.yaml) using [the following](#example-periodic-prow-job-config) example. Add it to the bottom of the file. The next step will move it per file format. 
+4. Run `make jobs` in release repo base directory.
+5. Commit the changes and make a PR.
+ 
+### Example Periodic Prow Job Config
+Defined in release repo: https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/osde2e/openshift-osde2e-main-periodics.yaml#L158 
+
+Job on Prow: https://prow.ci.openshift.org/?job=osde2e-rosa-stage-example-addon 
+
+Comments denote brief instructions. To adapt this to your job, you would redefine the values denoted with comments.
+Do not update keys with no comments next to them. For your job, do not copy from here, use the git version linked above instead. `ci-operator` does not like comments in yamls. 
+```yaml
+- agent: kubernetes
+  cluster: build05
+  cron: 0 14 * * 0 // runs once a week. update as needed. 
+  decorate: true
+  extra_refs:
+    - base_ref: main
+      org: openshift
+      repo: osde2e
+  labels:
+    pj-rehearse.openshift.io/can-be-rehearsed: "false"
+  name: osde2e-rosa-stage-example-addon  // update to your job in the format: osde2e-provider-environement-addon_name-addon
+  spec:
```

Comment:
We want to add the slack report for failied jobs:


```
  reporter_config:
    slack:
      channel: '#sd-cicd-alerts' // update to owner's channel.
      job_states_to_report:
      - failure
      - error
      report_template: 'Job {{.Spec.Job}} failed: {{.Status.URL}}'
```

Reply:
updated.. need a PR for release repo to add there